push/src/main/java/org/aerogear/mobile/push/PushService.java

```diff
@@ -0,0 +1,250 @@
+package org.aerogear.mobile.push;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.google.firebase.iid.FirebaseInstanceId;
+import com.google.firebase.messaging.FirebaseMessaging;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Base64;
+
+import org.aerogear.mobile.core.Callback;
+import org.aerogear.mobile.core.MobileCore;
+import org.aerogear.mobile.core.ServiceModule;
+import org.aerogear.mobile.core.configuration.ServiceConfiguration;
+import org.aerogear.mobile.core.exception.HttpException;
+import org.aerogear.mobile.core.http.HttpRequest;
+import org.aerogear.mobile.core.http.HttpResponse;
+
+public class PushService implements ServiceModule {
+
+    private static final String PUSH_CONFIG_FILE_NAME = "push-config.json";
+    private static final String JSON_OBJECT = "android";
+    private static final String JSON_SENDER_ID = "senderID";
+    private static final String JSON_VARIANT_ID = "variantID";
+    private static final String JSON_VARIANT_SECRET = "variantSecret";
+
+    private static final String registryDeviceEndpoint = "rest/registry/device";
+
+    private static final List<MessageHandler> MAIN_THREAD_HANDLERS = new ArrayList<>();
+    private static final List<MessageHandler> BACKGROUND_THREAD_HANDLERS = new ArrayList<>();
```

Comment:
Do these still need to be static?

Reply:
It's used for `notifyHandlers` and the method is static